### PR TITLE
Python 3.7 testing and compatibility

### PR DESCRIPTION
**Pull request by jklenzing**
Addresses #207.

Summary of changes:
- Updates Travis CI testing to use python 3.7 in the xenial environment
- Fixed a bug in python 3.7 caused by change in behvaiour of StopIteration (#207)
- Cleaned up unused imports and commented code.
- pep8 style adjustments

### CODE REVIEW

**Comment by jklenzing** [comment]
Since this changes the code and updates travis, I ran a version of this that tested 3.5 and 3.6 for the same code as well on Travis CI to verify compatibility across python versions:  https://travis-ci.org/rstoneback/pysat/builds/517592287

The final version will just test 2.7 and 3.7.

**Comment by jklenzing** [comment]
Pinging @aburrell for review

File: pysat/_orbits.py [hunk 1]
```diff
@@ -817,18 +816,9 @@ def __iter__(self):
...
-        #         self.sat.load(date=date)
-        #         break
-        # else:
-        #     raise ValueError('Iteration type not set')
-
         while True:
-            self.next()
-            yield self.sat
+            try:
+                self.next()
```
Comment (aburrell) [inline pysat/_orbits.py]: How did this exit previously?

Comment (jklenzing) [reply pysat/_orbits.py]: Previously, the `StopIteration` raised in the `.next()` would do exactly that.  With python 3.7, `StopIteration` raises a `RuntimeError` and crashes.  The new syntax is what other instances of the pysat iterators use, I think this is just something that slipped through the cracks.  We didn't see it until 3.7 forced us to.

**Pull request merged**